gulpfile.js/build.js

```diff
@@ -321,56 +353,57 @@ async function buildPages(done) {
 }
 
 /**
- * Transforms already built pages and does so while spawning multiple child
- * processes to speed up processing
+ * Removes unnecessary whitespace from rendered pages and minifies their CSS
  *
  * @return {Promise}
  */
-async function transformPages() {
-  let paths = await globby([
-    `${project.paths.GROW_BUILD_DEST}/**/*.html`,
-    `!${project.paths.GROW_BUILD_DEST}/{*/shared,shared}/*.html`,
-  ]);
-  let shardCount = os.cpus().length;
-  if (paths.length < shardCount) {
-    shardCount = paths.length;
-  }
-  const shardPathCount = Math.trunc(paths.length / shardCount);
-
-  // If there is no shard option it means this is the initial call to the task
-  // that spawns the subprocesses
-  if (config.options.shard === undefined) {
-    signale.info(`Spawning ${shardCount} processes to transform ${paths.length} pages ...`);
-    const shards = [];
-
-    while (shards.length < shardCount) {
-      const shardId = shards.length;
-      const shard = sh(`gulp transformPages --env ${config.environment} \
-          --shard ${shardId}`);
-      signale.success(`Started shard ${shardId} ...`);
-      shards.push(shard);
-    }
+function minifyPages() {
+  // Configure CleanCSS to use a more aggressive set of rules to achieve better
+  // results
+  const cleanCss = new CleanCSS({
+    2: {
+      all: true,
+      mergeSemantically: true,
+      restructureRules: true,
+    },
+  });
 
-    return Promise.all(shards);
-  }
+  // Only shorten selectors that are actually safe to be rewritten (i.e. are
+  // not used inside `amp-bind` statements for example)
+  rcs.selectorLibrary.setExclude(
+      new RegExp('^(?!' + SELECTOR_REWRITE_SETTINGS.safeSelectors.join('|') + ').*$')
+  );
+
+  return gulp.src(`${project.paths.GROW_BUILD_DEST}/**/*.html`)
+      .pipe(through.obj(async function(page, encoding, callback) {
+        let html = page.contents.toString();
+
+        // Compress multiple spaces down to only one
+        html = html.replace(/\s{2,}/gm, ' ');
+
+        // Extract the CSS in order to minify it
+        const css = html.match(/<style amp-custom>(.*?)<\/style>/ms);
+        if (css) {
+          const minifiedCss = cleanCss.minify(css[1]).styles;
+          html = html.replace(css[1], minifiedCss);
```

Comment:
this probably would be more efficient using indexes from the match instead of performing another string comparison.

Reply:
A little more verbose, but yes good idea!